releasenotes/notes/py310-support-869d47583c976eef.yaml

```diff
@@ -0,0 +1,17 @@
+---
+features:
+  - |
+    Added support for running with Python 3.10. This includes publishing
+    precompiled binaries to PyPI for Python 3.10 on supported platforms.
+upgrade:
+  - |
+    Qiskit no longer fully supports 32 bit platforms, Linux i686 and Windows
+    32bit, on Python >= 3.10. This is because the upstream dependencies Numpy
+    and Scipy do not support these platforms on Python 3.10 and we're unable to
+    reliably test on these platforms anymore. Qiskit will still publish
+    precompiled binaries for these platforms but we're unable to test them
+    prior to publishing and you will need a C/C++ compiler to install them
+    as numpy, scipy, and potentially other dependencies will need to be built
+    from source when installing qiskit-terra on Python 3.10 on linux i686 or
+    win32. If you're running on i686 or win32 platforms it is recommended that
+    you use Python 3.7, 3.8, or 3.9.
```

Comment:
In the current form it's a little hard to read - the sentences are quite long, and we're a bit inconsistent with capital letters/spacing/other stylist things.  Perhaps something like

> Qiskit no longer fully supports 32-bit platforms on Python >= 3.10. These are Linux i686 and 32-bit Windows. This is because the upstream dependencies Numpy and Scipy have also dropped support for them. Qiskit will still publish precompiled binaries for these platforms, but we are unable to test the packages prior to publishing, and you will need a C/C++ compiler so that ``pip`` can build their dependencies from source.  If you are using one of these platforms, we recommend that you use Python 3.7, 3.8, or 3.9.

Reply:
Yeah, this is probably because it's pretty nuanced in what this change means and also because I kind of changed directions at the last minute in writing this (originally I just dropped 32bit platforms completely but then decided it would be better to still build the wheels and just not test them). To make this a bit easier to understand I just pushed https://github.com/Qiskit/qiskit/pull/1402 to define our supported platform matrix a bit better. Once that merges I can basically say for Python >=3.10 on i686 and win32 it's Tier 3 support instead of Tier 2.

I like your suggested text, it's clearer than what I came up with. I'll incorporate that in with the tiers when I update this.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7553/commits/07f6917246371f454a6ab10792eda9ddf209b8ad